distribution/client/src/lib/update.ts

```diff
@@ -79,24 +87,41 @@ export default class Update {
     return this.app.service('update/tainted').create({
       uuid: this.uuid,
       level: toBeTaintedLevel
-    }, {});
+    }, {})
+    .catch( (e) => {
+      logger.error('Tainting went wrong!', e);
+      throw e;
+    });
   }
   /*
-   * Apply the updates provided by the given Update Manifest
-   *
+   * Apply the updates provided by the given Update Manifest.
+   * Rolls back automatically to the previous non tainted level if failing.
+   * (Note: the rollback may not end up on most of the time, that would mean going from say UL-0 to U)
+
+   1) UL10 gets globally tainted after an issue is identified (UL9 is not tainted)
+   2) Instance attempts updating from UL10 to UL11, and fails
+   3) update.query() to get the right updates => receives UL9...
+   4) config files changed from UL9 to UL10, Jenkins fails to start after rolling back.
+
    * @param  {Map} Update Manifest described in JEP-307
    * @return {Promise} Which resolves once updates have been applied
    * @return {boolean} False if there is already an update in progress
    */
-  async applyUpdates(updates) {
-    if (this.updateInProgress || (!updates)) {
+  async applyUpdates(updates, forceUpdate?: boolean) {
+    if (forceUpdate) {
+      logger.warn('Forced update (expected during a rollback)');
+    }
+    else if (this.updateInProgress || (!updates)) {
       logger.warn('applyUpdates request ignored: update already in progress!');
       return false;
+    } else {
+      // Setting this to a timestamp to make a timeout in the future
+      this.updateInProgress = new Date();
     }
+    const applyUpdatesMessage = 'Starting to apply updates';
+    logger.info(applyUpdatesMessage)
+    UI.publish(applyUpdatesMessage);
```

Comment:
This can be made into a single call, and we can remove the temporary variable with:

`UI.publish('Starting to apply updates', { log: 'info' })`